src/mame/drivers/ddealer.cpp

```diff
@@ -401,12 +390,22 @@ TIMER_DEVICE_CALLBACK_MEMBER(ddealer_state::mcu_sim)
 
 
 
-void ddealer_state::back_vram_w(offs_t offset, uint16_t data, uint16_t mem_mask)
+void ddealer_state::back_vram_w(offs_t offset, u16 data, u16 mem_mask)
 {
 	COMBINE_DATA(&m_back_vram[offset]);
 	m_back_tilemap->mark_tile_dirty(offset);
 }
 
+void ddealer_state::fg_vram_w(offs_t offset, u16 data, u16 mem_mask)
+{
+	COMBINE_DATA(&m_fg_vram[offset]);
+	m_fg_tilemap->mark_tile_dirty(offset);
+	if (offset & 0x800)
+		m_fg_tilemap_right->mark_tile_dirty(offset & ~0x800);
+	else
+		m_fg_tilemap_left->mark_tile_dirty(offset & ~0x800);
```

Comment:
Do you really want to mask out a single bit here, or do you want to get the low eleven bits (i.e. `offset & 0x7ff`)?

Reply:
Bit 12 is select top/bottom 256 pixel area in the foreground tilemap, so topmost 1 bit is needed for tilemap.
see TILEMAP_MAPPER_MEMBER(ddealer_state::scan_fg).